core/state/state_object.go

```diff
@@ -150,20 +155,20 @@ func (s *stateObject) touch() {
 	}
 }
 
-func (s *stateObject) getTrie(db Database) Trie {
+func (s *stateObject) getTrie() Trie {
 	if s.trie == nil {
 		// Try fetching from prefetcher first
 		// We don't prefetch empty tries
 		if s.data.Root != emptyRoot && s.db.prefetcher != nil {
 			// When the miner is creating the pending state, there is no
 			// prefetcher
-			s.trie = s.db.prefetcher.trie(s.data.Root)
+			s.trie = s.db.prefetcher.trie(s.addrHash, s.data.Root)
 		}
 		if s.trie == nil {
 			var err error
-			s.trie, err = db.OpenStorageTrie(s.addrHash, s.data.Root)
+			s.trie, err = s.db.db.OpenStorageTrie(s.db.originalRoot, s.addrHash, s.data.Root)
 			if err != nil {
-				s.trie, _ = db.OpenStorageTrie(s.addrHash, common.Hash{})
+				s.trie, _ = s.db.db.OpenStorageTrie(s.db.originalRoot, s.addrHash, common.Hash{})
```

Comment:
I think we should handle error here, because `OpenStorageTrie` will return nil if failed to create new sercuredTrie.
Try to cherry-pick this commit, it's merged to geth already

https://github.com/rjl493456442/go-ethereum/commit/01808421e20ba9d19c029b64fcda841df77c9aff#diff-493fcc97ac5eefa37eef088731c482655afbae6c1ebaead72a88b7746897c606R164

Reply:
Linked isssue: https://github.com/ethereum/go-ethereum/pull/26350